pkg/extract/zip.go

```diff
@@ -0,0 +1,116 @@
+package extract
+
+import (
+	"archive/zip"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/replicate/pget/pkg/logging"
+)
+
+// ZipFile extracts a zip file to the given destination path.
+func ZipFile(reader io.ReaderAt, destPath string, size int64, overwrite bool) error {
+	logger := logging.GetLogger()
+	err := os.MkdirAll(destPath, 0755)
+	if err != nil {
+		return fmt.Errorf("error creating destination directory: %w", err)
+	}
+
+	logger.Debug().
+		Str("extractor", "zip").
+		Str("status", "starting").
+		Bool("overwrite", overwrite).
+		Str("destDir", destPath).
+		Msg("Extract")
+	zipReader, err := zip.NewReader(reader, size)
+	if err != nil {
+		return fmt.Errorf("error creating zip reader: %w", err)
+	}
+
+	for _, file := range zipReader.File {
+		err := handleFileFromZip(file, destPath, overwrite)
+		if err != nil {
```

Comment:
extracted files do not end up outside the intended destination directory?
I assume ZIP code checks archive's size, structure for signs of corrupted/junk archive...
same for other arch types.

Reply:
Malicious Tar and Zip checking should be added.